mautic/mautic
# Ensure all form fields are included within the form tag

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | N
| Automated tests included? | Y
| Related user documentation PR URL | N/A
| Related developer documentation PR URL | N/A
| Issues addressed (#s or URLs) | /
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

~~⚠️ this PR needs a service from https://github.com/mautic/mautic/pull/10846 so this cannot be merged without merging that PR first.~~

⚠️ There is a followup ticket https://github.com/mautic/mautic/pull/11443 that fixes other issues, but this PR is big already so let's get it merged and then continue with that one.

[//]: # ( Required: )
#### Description:
This PR fixes how we're wrapping form elements on the unsubscribe / preference center page. 

In order to facilitate testing and readability, I also included many adjustments and refactorings to private methods within the BuilderSubscriber. 

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Create an email with a custom preference center landing page
2. Send that email through a campaign to a user
3. Click unsubscribe and fill out the form.
4. On submit, you should receive a `CSRF token is invalid` error message

#### Steps to test this PR:
1. Redo reproduction steps
2. When you submit, the form should submit successfully
3. To test the refactoring, ensure that your preference center page has all available parts - frequency rules, segments, channels, etc.
